src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -723,6 +723,61 @@ std::vector<stan::math::var> to_vars__(std::initializer_list<stan::math::var> x)
   return x;
 }
 
+template <typename T>
+void FnCheckShape__(const char* var_name, size_t depth,
+                    const T& obj) {
+}
+
+template <typename T>
+void FnCheckShape__(const char* var_name, size_t depth,
```

Comment:
Could we move this functions to the stan-dev/stan repo? Not sure of the best location but I guess under /model/check.hpp and then add that header to the model_header? They would probably have to be named more in the style of math / stan C++ conventions we use. So something like `check_shape()`.

The other functions that dont depend on model specific should probably also go there but that is not this PRs fault and most of them can actually just be removed.

Reply:
Yeah, I was definitely being lazy. I think this actually belongs in the math library. FnCheckShape__ is a generalization of [check_consistent_size](https://github.com/stan-dev/math/blob/develop/stan/math/prim/err/check_consistent_size.hpp). Alternatively it could be replaced by [check_matching_dims](https://github.com/stan-dev/math/blob/develop/stan/math/prim/err/check_matching_dims.hpp) if that recursed into arbitrary arrays.

Reply:
I am fine with either of the proposed solutions.